scalalib/worker/src/mill/scalalib/worker/ZincProblem.scala

```diff
@@ -15,4 +17,7 @@ class ZincProblem(base: xsbti.Problem) extends Problem {
   override def message: String = base.message()
 
   override def position: ProblemPosition = new ZincProblemPosition(base.position())
+
+  override def diagnosticCode: Option[DiagnosticCode] =
+    base.diagnosticCode().toScala.map(ZincDiagnosticCode)
```

Comment:
So one thing I don't fully understand and haven't chased down yet is that this is currently `None` no matter what. Even when I am using my self published snapshot of Scala 3 with the changes in https://github.com/lampepfl/dotty/pull/15565 I still don't get the code here even though I have tested that it is passed into `Problem`. So I'm doing something wrong, but still not sure what. If anything on this end looks wrong, just let me know. If not I'll just keep digging.